pandas/tests/frame/test_indexing.py

```diff
@@ -3042,6 +3044,20 @@ def test_transpose(self):
         expected.index = ['A', 'B']
         assert_frame_equal(result, expected)
 
+    def test_scalar_assignment(self):
+        # issue #19843
+        df = pd.DataFrame(index=(0, 1, 2))
+        df['now'] = pd.Timestamp('20130101', tz='UTC')
+        assert isinstance(df.dtypes[0], DatetimeTZDtype)
```

Comment:
construct the expected frame and use ``assert_frame_equal``

Reply:
put this in ``class TestDataFrameIndexingDatetimeWithTZ(TestData)``

name it ``test_setitem_scalar``

parameterize it with both a tz and a tz-aware value.